### PR TITLE
#191: Renamed `parameter` to `parameters` to match JSON input in usrr guide

### LINKED ISSUE #191: Rename UDF json element "parameters" to "parameter"
Currently, the example in the user guide specifies plural `"parameters": "bla-bla"` while the constructor of the example UDF class contains an argument `parameter` (singular). 

In json you probably can specify other datatypes such as int, float or boolean as well as more complex data, such as a json object or an array while the argument of the constructor of the UDF class still remains a string.

Probably, the json value is converted to a string and the UDF constructor can optionally use json library to get back other datatypes. 

However, the argument's numerus should be identical.

The current ticket requests to name it to singular. This requires updating 
* the user guide
* the lua script [query_loop.lua](https://github.com/exasol/advanced-analytics-framework/blob/main/exasol_advanced_analytics_framework/lua/src/query_loop.lua).

### PR DESCRIPTION
**Pull request by ckunki**
Closes  #191

### CODE REVIEW

File: tests/test_package/test_query_handlers/query_handler_test.py [hunk 1]
```diff
@@ -99,14 +99,14 @@ def handle_query_result(self, query_result: QueryResult) -> Union[Continue, Fini
 
 class QueryHandlerWithOneIterationWithNotReleasedChildQueryHandlerContextFactory(UDFQueryHandlerFactory):
 
-    def create(self, parameter: str, query_handler_context: ScopeQueryHandlerContext) -> UDFQueryHandler:
-        return QueryHandlerWithOneIterationWithNotReleasedChildQueryHandlerContext(parameter, query_handler_context)
+    def create(self, parameters: str, query_handler_context: ScopeQueryHandlerContext) -> UDFQueryHandler:
+        return QueryHandlerWithOneIterationWithNotReleasedChildQueryHandlerContext(parameters, query_handler_context)
```
Comment (ahsimb) [inline tests/test_package/test_query_handlers/query_handler_test.py:103]: reminds me a Welsh train station name:
Llanfairpwllgwyngyllgogerychwyrndrobwllllantysiliogogogoch

**Review by ahsimb — APPROVED** [review]

File: tests/test_package/test_query_handlers/query_handler_test.py (hunk 1 above)
Comment (tkilias) [reply tests/test_package/test_query_handlers/query_handler_test.py:103]: This is actually a test case, thats why we need the lengthy name to be precise

**Review by tkilias — CHANGES_REQUESTED** [review]
We wanted to go with parameter and not parameters.

File: tests/test_package/test_query_handlers/query_handler_test.py (hunk 1 above)
Comment (ckunki) [reply tests/test_package/test_query_handlers/query_handler_test.py:103]: :rofl:

**Pull request closed without merging**